docs/pages/guides/upgrading/0.19.mdx

```diff
@@ -0,0 +1,220 @@
+  <App />
+</RoomProvider>
+```
+
+For additional context, see the
+[troubleshooting guide](/docs/guides/troubleshooting#stale-props-zombie-child)
+
+### New shouldInitiallyConnect prop [#react-shouldInitiallyConnect]
+
+We added a new property `shouldInitiallyConnect` to `RoomProvider`, which lets
+you control whether or not the room connects to Liveblock servers. By default,
+it will check the `typeof window` to determine if it should connect. When using
+SSR, you can set it to `false` to prevent the room from connecting to Liveblocks
+servers.
+
+```tsx highlight="4"
+<RoomProvider
+  id="room"
+  initialPresence={{}}
+  shouldInitiallyConnect={false} // 👈 control initial connection to Liveblocks
+>
+  <App />
+</RoomProvider>
+```
+
+### Addition of @liveblocks-core package [#liveblocks-core]
+
+We restructured our internal packages to increase code sharing. You may notice a
+new dependency in your dependency tree: `@liveblocks/core`. It contains private
+APIs that aren't intended for direct consumption.
+
+## Changes in @liveblocks/client [#client]
+
+Similar to the [` shouldInitiallyConnect prop`](#react-shouldInitiallyConnect)
+in `@liveblocks/react`, we added a `shouldInitiallyConnect`option
+to`client.enter`. The default value is `true`. You may want to set it to `false`
+when using SSR, or if you would prefer to establish the WebSocket connection
+later.
+
+```tsx
+client.enter("room", {
+  initialPresence: {},
+  shouldInitiallyConnect: false, // 👈 for SSR using the @liveblocks/client package
+});
+```
+
+## Changes in @liveblocks/redux [#redux]
+
+### liveblocksEnhancer is now the default export [#liveblocksEnhancer]
+
+The main export has been renamed, so you will need to update your imports and
+use of the enhancer:
+
+#### Previously ❌ [@hidden]
+
+`import { enhancer } from "@liveblocks/redux";`
+
+#### Now ✅ [@hidden]
+
+`import { liveblocksEnhancer } from "@liveblocks/redux";`
+
+### enterRoom now accepts a second argument to set the initial state [#enterRoom]
+
+The second argument to `state.liveblocks.enterRoom()` to send in an explicit
+initial state is no longer supported. It will use the state in your Redux store,
+for consistency and ease of use.
+
+#### Previously ❌ [@hidden]
+
+```ts
+useEffect(() => {
+  enterRoom("room-id", {
+    todos: [],
+  });
+});
+```
+
+#### Now ✅ [@hidden]
+
+```ts
+useEffect(() => {
+  enterRoom("room-id");
+});
+```
+
+## Changes in @liveblocks/zustand [#zustand]
+
+In 0.19 we added support for Zustand v4 (specifically v4.1.3 or higher) and will
+no longer support Zustand v4.1.2 or lower. This is because Zustand v4.1.3 brings
+greatly improved TypeScript types to its APIs, and consequently, we can improve
+our internal types. To migrate, make the following code changes:
+
+- `npm update @liveblocks/client@0.19 @liveblocks/zustand@0.19`
+- Change these imports, if applicable:
+  ```diff
+  -import { middleware } from "@liveblocks/zustand";
+  +import { liveblocks } from "@liveblocks/zustand";
```

Comment:
I don't find the -import +import very clear. And from what I see in our repo, we have never communicated like that before

Why don't we do a before and after box like usual?